src/meta-srv/src/lock/memory.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use async_trait::async_trait;
+use tokio::sync::{Mutex, OwnedMutexGuard};
+
+use crate::error::Result;
+use crate::lock::{DistLock, Key, Opts};
+
+#[derive(Default)]
+pub(crate) struct MemLock {
+    mutexes: Arc<Mutex<HashMap<Key, Arc<Mutex<()>>>>>,
```

Comment:
Looks like Arc can be removed.
```suggestion
    mutexes: Mutex<HashMap<Key, Arc<Mutex<()>>>>,
```